TTT/AI.cpp

```diff
@@ -0,0 +1,54 @@
+#include "AI.hpp"
+
+AI::AI(GameField::Value sign, GameField::Value opponent)
+	: m_sign(sign), m_opponent(opponent), m_name("AI") {};
+
+void AI::playerMove(GameField& field) {
+	int y, x;
+	if (aiWinChk(field)) return;
+	if (humWinChk(field)) return;
+
+	std::random_device random;
+	std::mt19937_64 mt(random());
+	std::uniform_real_distribution<double> dist(0, field.winSeq());
+
+	do {
+		y = dist(mt);
+		x = dist(mt);
+	} while (!field.isEmpty(y, x));
+	field.setValue(y, x, m_sign);
```

Comment:
You could just return the target cell and thus keep the field argument constant.